src/data/polynomial/degree/erase_leading.lean

```diff
@@ -0,0 +1,166 @@
+/-
+Copyright (c) 2020 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+import data.polynomial.degree.basic
+import data.polynomial.degree.to_basic
+import data.polynomial.degree.to_trailing_degree
+import data.polynomial.degree.trailing_degree
+
+open polynomial finsupp finset
+
+namespace polynomial
+
+variables {R : Type*} [semiring R] {f : polynomial R}
+
+namespace erase_leading
+
+/-- erase_lead of a polynomial f is the polynomial obtained by
+subtracting from f the leading term of f. -/
+def erase_lead (f : polynomial R) : polynomial R := ⟨ f.support \ singleton f.nat_degree , λ a : ℕ , ite (a = f.nat_degree) 0 f.coeff a , λ a , begin
```

Comment:
Seeing as we already have `finsupp.erase` used in the polynomial library, I'd like to be consistent with that, and use `finsupp.erase` to implement this function. I know you expressed some qualms about the computability, so I'd like to summon someone more knowledgeable than myself to judge this. My impression, though, was that in the polynomial library, we've kind of thrown all computability out the window, so we shouldn't worry.

Reply:
Ok, I will wait a little bit longer to see if someone else chimes in.  Otherwise, I will switch to noncomputable.